src/Microsoft.Crank.Agent/Startup.cs

```diff
@@ -923,11 +923,29 @@ private static async Task ProcessJobs(string hostname, string dockerHostname, Ca
 
                                             if (job.State != JobState.Failed && benchmarksDir != null)
                                             {
-                                                process = await StartProcess(hostname, Path.Combine(tempDir, benchmarksDir), job, _dotnethome, context);
+                                                try
+                                                {
+                                                    process = await StartProcess(hostname, Path.Combine(tempDir, benchmarksDir), job, _dotnethome, context);
 
-                                                Log.Info($"Process started: {job.ProcessId}");
+                                                    Log.Info($"Process started: {job.ProcessId}");
 
-                                                workingDirectory = process.StartInfo.WorkingDirectory;
+                                                    workingDirectory = process.StartInfo.WorkingDirectory;
+                                                }
+                                                catch (Exception e)
+                                                {
+                                                    job.Error = "Error while starting the application: " + e.ToString();
+                                                    Log.Info(job.Error);
+
+                                                    if (job.State != JobState.Deleting &&
+                                                        job.State != JobState.Deleted &&
+                                                        job.State != JobState.Failed &&
+                                                        job.State != JobState.Stopped
+                                                    )
+                                                    {
```

Comment:
Worth logging the exception here?

Reply:
It's logged on line 937 (server-side) and reported to the controller on 936.